src/app/ReadClient.cpp

```diff
@@ -720,5 +753,42 @@ CHIP_ERROR ReadClient::SendSubscribeRequest(ReadPrepareParams & aReadPreparePara
     return CHIP_NO_ERROR;
 }
 
-}; // namespace app
-}; // namespace chip
+void ReadClient::OnResubscribeTimerCallback(System::Layer * apSystemLayer, void * apAppState)
+{
+    ReadClient * const client = reinterpret_cast<ReadClient *>(apAppState);
+    if (client != nullptr)
+    {
+        client->SendSubscribeRequest(client->mReadPrepareParams);
+        client->mNumRetries++;
+    }
+    else
+    {
+        ChipLogError(DataManagement, "Client is null");
+    }
+}
+
+CHIP_ERROR ReadClient::Resubscribe()
+{
+    mReadPrepareParams.mpResubscribeDelegate->Run(mNumRetries, mIntervalMsec, mReadPrepareParams.mShouldResubscribe);
+    if (!mReadPrepareParams.mShouldResubscribe)
```

Comment:
See comment below - a temporary variable should be sufficient.

Reply:
updated. thanks